ingestion/portland_ingest.py

```diff
@@ -21,131 +21,164 @@
+    reads and individual Excel file and returns an error
+    if the file path does not exist
     """
     try:
-        # load data sets
-        spring_data_sat = pd.read_excel(folder_path + spring_sat)
-        spring_data_week = pd.read_excel(folder_path + spring_weekday)
-        spring_data_sun = pd.read_excel(folder_path + spring_sunday)
-
-        summer_data_sat = pd.read_excel(folder_path + summer_sat)
-        summer_data_week = pd.read_excel(folder_path + summer_weekday)
-        summer_data_sun = pd.read_excel(folder_path + summer_sunday)
-
-        # add flags to specify season and day of week, this will be helpful
-        # when converting the dataset from average by season to day level
-        spring_data_sat["day_type"] = "Saturday"
-        spring_data_week["day_type"] = "Weekday"
-        spring_data_sun["day_type"] = "Sunday"
-
-        summer_data_sat["day_type"] = "Saturday"
-        summer_data_week["day_type"] = "Weekday"
-        summer_data_sun["day_type"] = "Sunday"
-
-        spring_data_sat["season"] = "Spring"
-        spring_data_week["season"] = "Spring"
-        spring_data_sun["season"] = "Spring"
-
-        summer_data_sat["season"] = "Summer"
-        summer_data_week["season"] = "Summer"
-        summer_data_sun["season"] = "Summer"
-
-        # append into one data set
-        all_rider_data = pd.concat(
-            [
-                spring_data_sat,
-                spring_data_week,
-                spring_data_sun,
-                summer_data_sat,
-                summer_data_week,
-                summer_data_sun,
-            ],
-            ignore_index=True,
-        )
-
-        print("Data loaded and appended successfully.")
-
+        return pd.read_excel(file_path)
     except FileNotFoundError:
-        print("Files not found. Check the file path and filename.")
+        print("File not found:", file_path)
     except Exception as e:
-        print("An error occurred:", e)
-
-
-# load data sets
-all_rider_data = load_ridership_dasets()
-
-# format data sets
-all_rider_data.columns = all_rider_data.columns.str.lower().str.replace(" ", "_")
-
-all_rider_data = all_rider_data[["location_id", "ons", "day_type", "season"]]
-
-# Since the data is at the day type + season level,
-# create a df with all calendar days of 2023
-start_date = "2023-01-01"
-end_date = "2023-12-31"
-calendar_dates = pd.date_range(start=start_date, end=end_date, freq="D")
+        print("An error occurred while loading data:", e)
 
-# flag days of 2023 to match day_type field
-calendar_df = pd.DataFrame(calendar_dates, columns=["date"])
 
-calendar_df["day_type"] = calendar_df["date"].dt.day_name()
-
-day_type_mapping = {"Saturday": "Saturday", "Sunday": "Sunday"}
-
-calendar_df["day_type"] = (
-    calendar_df["day_type"].map(day_type_mapping).fillna("Weekday")
-)
+def load_ridership_datasets(file_names):
+    """
+    Takes a dictionary in the form key: season_day type
```

Comment:
This is an example where I think type hints could be quite helpful as "file_names" is ambiguous and its easier to check with type hints what is expected.